os/arch/xtensa/src/esp32/esp32_i2c.c

```diff
@@ -335,20 +354,31 @@ static int i2c_esp32_isr(int irq, FAR void *context, FAR void *arg)
 				sem_post(&priv->sem_isr);
 				break;
 			} else if (status & I2C_END_DETECT_INT_ST_M) {
+				I2C[priv->i2c_num]->int_ena.end_detect = 0;
 				I2C[priv->i2c_num]->int_clr.end_detect = 1;
+				if (i2c_esp32_check_bus(priv) == ESP32_BUS_BUSY) {
+					priv->status = I2C_STATUS_TIMEOUT;
```

Comment:
This case, it failed because of i2c state is busy but returns OK, is it correct flow? I don't think so.

Reply:
it will use priv->status to notice i2c_esp32_transmit, it will recieve I2C_STATUS_TIMEOUT then returns  -ETIMEDOUT. please check it.

Reply:
@xiarihf 
you mean that is it correct to setup hsi2c although i2c state is busy?

Reply:
Sorry, i did not catch your question. do you mean hsi2c_setup function? it is only called in driver initializing in booting period.  The i2c bus is transmitting data, and the status reg bit I2C_BUS_BUSY is 1, after it transmitting data, it will be 0.